### PR TITLE
Feature bugsnag, show log and catch exception

### PR DESCRIPTION
**Pull request by rafaelje**
## Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: [Conventional Commit](http://conventionalcommits.org/)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue # N/A

## What is the new behavior?
- Add bugsnag
- Show log when installing as root and without SDCard
- Catch exception when start

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: app/src/main/java/org/flyve/mdm/agent/utils/UtilsCrash.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * @link      https://github.com/flyve-mdm/flyve-mdm-android
+ * @link      https://flyve-mdm.com
+ * ------------------------------------------------------------------------------
+ */
+public class UtilsCrash {
+
+    public static void configCrash(Context context, boolean val) {
+        if(val) {
+            Bugsnag.init(context);
+            Bugsnag.setEndpoint("https://hooks.thestralbot.com/-1001061475099/");
```
Comment (ajsb85) [inline app/src/main/java/org/flyve/mdm/agent/utils/UtilsCrash.java]: Could you get this URL by an env var in Travis CI?

Comment (rafaelje) [reply app/src/main/java/org/flyve/mdm/agent/utils/UtilsCrash.java]: Yes I can

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -56,6 +56,10 @@
         android:label="@string/app_name"
         android:theme="@style/AppTheme">
 
+        <meta-data
+            android:name="com.bugsnag.android.API_KEY"
+            android:value="b6bc7ce0da33f92fe6c49c568af43963"/>
```
Comment (ajsb85) [inline app/src/main/AndroidManifest.xml]: Could you get this API Key by an env var in Travis CI?

File: app/src/main/java/org/flyve/mdm/agent/ui/MDMAgent.java [hunk 3]
```diff
@@ -32,14 +32,14 @@
...
 import android.support.multidex.MultiDex;
-import android.util.Log;
 
 import com.orhanobut.logger.AndroidLogAdapter;
 import com.orhanobut.logger.FormatStrategy;
 import com.orhanobut.logger.Logger;
 import com.orhanobut.logger.PrettyFormatStrategy;
 
 import org.flyve.mdm.agent.data.DataStorage;
+import org.flyve.mdm.agent.utils.UtilsCrash;
```
Comment (ajsb85) [inline app/src/main/java/org/flyve/mdm/agent/ui/MDMAgent.java]: DRY

File: app/src/main/java/org/flyve/mdm/agent/ui/PreviewSupervisorActivity.java [hunk 4]
```diff
@@ -9,6 +9,7 @@
 import org.flyve.mdm.agent.R;
 import org.flyve.mdm.agent.core.supervisor.SupervisorController;
 import org.flyve.mdm.agent.core.supervisor.SupervisorModel;
```
Comment (ajsb85) [inline app/src/main/java/org/flyve/mdm/agent/ui/PreviewSupervisorActivity.java]: DRY

File: app/src/main/java/org/flyve/mdm/agent/utils/FlyveLog.java [hunk 5]
```diff
@@ -41,7 +44,7 @@
 
     private static final String FILE_NAME_FEEDBACK = "FlyveMDMFeedback.txt";
     public static final String FILE_NAME_LOG = "FlyveMDMLog.txt";
-    public static final String FLYVE_PATH = Environment.getExternalStorageDirectory().getPath() + "/FlyveMDM";
+    public static final String FLYVE_PATH = Environment.getExternalStorageDirectory().getPath() + "/FlyveMDM/";
```
Comment (ajsb85) [inline app/src/main/java/org/flyve/mdm/agent/utils/FlyveLog.java]:
```JAVA
FLYVE_PATH = Environment.getExternalStorageDirectory().getAbsolutePath() + File.separator + "FlyveMDM";

```

File: app/src/main/assets/setup.properties [hunk 6]
```diff
@@ -1,2 +1,3 @@
 #
-setup.admin_web_console=http://demo.flyve.org
\ No newline at end of file
+setup.admin_web_console=http://demo.flyve.org
+setup.thestralbot=http://demo.flyve.org
```
Comment (ajsb85) [inline app/src/main/assets/setup.properties]: https://hooks.thestralbot.com/-1001061475099/

Comment (rafaelje) [reply app/src/main/assets/setup.properties]: this is a fake url, the real url is in circleci

**Pull request merged**